6/6.go

```diff
@@ -0,0 +1,64 @@
+package main
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"strings"
+)
+
+func readFile(name string) ([]string, error) {
+	content, err := os.ReadFile(name)
+	if err != nil {
+		log.Fatalf("Nelze načíst soubor: %v", err)
+		return nil, err
+	}
+
+	groups := strings.Split(string(content), "\n\n")
+	return groups, nil
+}
+
+func removeDuplicates(s string) string {
+	seen := make(map[rune]bool)
+	result := []rune{}
+
+	for _, char := range s {
+		if !seen[char] && char != '\n' {
+			seen[char] = true
+			result = append(result, char)
+		}
+	}
+	return string(result)
+}
```

Comment:
Tady je nejspíš vágně pojmenovaná funkce – `removeDuplicates`, ale duplicates čeho? Chápal bych, kdyby dostávala pole něčeho a vracela pole toho stejného, ale bez duplicity, ale když dostává a vrací string, není úplně jasné, co to dělá. Pokud to ze stringu maže `char`y, které se tam vyskytují několikrát, tak bych to pojmenoval třeba `removeCharDuplicates` nebo nějak podobně.

Reply:
Rozumím